### ISSUE TITLE
Subgraphs fail when no traces are returned for a block

### ISSUE THREAD

**Issue by Jannis** [graphprotocol/graph-node#924]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

When a subgraph reaches the head of the chain, it can (or will?) run into a block for which the traces are not yet available. In this case, the subgraph will fail with an error like this one:

```
Subgraph instance failed to run: Trace stream return no traces for block: number = `7728630`, hash = `0xb1b1…648f`, code: SubgraphSyncingFailure, id: QmVozgjxwKKJWcDmbvv4AEWCj9jDbeGYyAqwiQqawJZMu8, updates: 0
```

Before that the subgraph already seems to try fetching the traces for the same block over and over again:

```
...
Requesting traces for block 7728630, component: BlockStream, updates: 0
Requesting traces for block 7728630, component: BlockStream, updates: 0
Requesting traces for block 7728630, component: BlockStream, updates: 0
Requesting traces for block 7728630, component: BlockStream, updates: 0
```

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem.**

1. Clone https://github.com/Jannis/gravity-subgraph
2. Deploy it to a Graph Node that is connected to a Parity archive node with tracing enabled and the "traces" API enabled
3. Wait for the subgraph to sync all the way, then wait a bit longer

**What is the expected behavior?**

I'm assuming the expected behavior is that the subgraph handles the absence of traces gracefully and keeps retrying?

**Comment by ziegfried** [comment]
Is there any workaround when deploying to api.thegraph.com? This seems to happen consistently when I try to update this subgraph: https://thegraph.com/explorer/subgraph/ziegfried/peepeth

**Comment by Jannis** [comment]
@ziegfried Not yet but we're working on a fix.

**Comment by Jannis** [comment]
Resolved as of #947.
